env_setup/env_setup.sh

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env bash
+
+# tput codes
+RST="$(tput sgr0)"
+COL_BLACK="$(tput setaf 0)"
+COL_RED="$(tput setaf 1)"
+COL_GREEN="$(tput setaf 2)"
+COL_YELLOW="$(tput setaf 3)"
+COL_BLUE="$(tput setaf 4)"
+COL_WHITE="$(tput setaf 7)"
+TXT_UNDERLINE="$(tput smul)"
+
+
+if [[ "$OSTYPE" == "darwin"* ]]; then
+    echo -e "$TXT_UNDERLINE---> MacOS detected so moving further with env_setup script. <---$RST\n"
+else
+    # exiting for now but later it can be updated
+    echo "Currently, script only works with MacOS!!!!!"
+    exit 1
+fi
+
+install_virt_pckgs() {
+    pip3 install virtualenv >&2 > /dev/null
+    pip3 install virtualenvwrapper >&2 > /dev/null
+}
+
+
+# echo "Do you to perform pre_setup?"
+# echo -e "!!!! PRE_SETUP:\nThis will make python3 and pip3 as default versions\n\t+\nit will install virtualenv and virtualenvwrapper\n!!!!\n"
+# echo -e "-- Enter the number for your selection and press enter!"
+# select yn in "Yes" "No"; do
+#     case $yn in
+#         Yes ) bash $(echo $PWD)/pre_setup.sh; break;;
+#         No ) echo -e "You selected \"NO\" so make sure that you have python3 and pip3 installed otherwise script will fail"; exit;;
+#     esac
+# done
+
+
+add_virtualenvwrppaer_env_var() {
+    FILE=$1
+    CONTENTS=$2
+    echo -e "$CONTENTS" >> $FILE
+}
+
+install_virt_pckgs
+
+### update environement vars
+curr_term=$(echo $SHELL)
+PROFILE_FILE=""
+echo "$curr_term"
+if [ "$curr_term" == "/bin/zsh" ]; then
+    PROFILE_FILE="$HOME/.zshrc"
+else
+    PROFILE_FILE="$HOME/.bash_profile"
+fi
```

Comment:
The user's system will automatically activate the necessary profile based on a shell. Do we really need this piece of code?

Reply:
In case, the developer wants to do it manually then one can avoid running the scripts. I created a script just for making things automated.
What do you think, should we remove this?

Reply:
We can keep it for now. No worries.